### DISCUSSION TITLE
Problem with frequency or bleeding? pls someone help me...😅😅

### DISCUSSION THREAD

**Discussion by jfbvm** [mrcodetastic/ESP32-HUB75-MatrixPanel-DMA#489] [Q&A]
Hi to every one,

I'm facing a problem that I'm not sure about its root cause, let me explain.

I have a display with 3x3 of 32x16 modules, but as you can see below there are some LEDs that were not to be on:

https://github.com/mrfaptastic/ESP32-HUB75-MatrixPanel-DMA/assets/3219805/429497c2-3122-4b1f-9334-8fc1d3c8f137

![image](https://github.com/mrfaptastic/ESP32-HUB75-MatrixPanel-DMA/assets/3219805/daf20614-c6fb-4893-af2a-da684c04c78e)

Its all over the display.

I'm using a power supply of and the AC GND is connected:
- In 127 Vac
- Out 5V
- Power 200W

For the code I'm using:
Default GPIO pins
i2sspeed: HUB75_I2S_CFG::HZ_20M (Already tried with others freqs but the problem only get worse)
min_refresh_rate: 30 (Already tried omitting this option and setting it, but nothing changed)
clkphase: false (Already tried with true and started to show wrong colors)

For now, I don't know what to do to solve this, pls someone help me...😅😅

**Comment by mrcodetastic** [comment]
Do you get the same issue when testing each panel individually when connected to the ESP32?

Looks like some electrical noise type issue.

**Comment by jfbvm** [comment]
Yes, already tested, the same issue...

Obter o Outlook para Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by daveythacher** [comment]
This is called ghosting. It occurs due to charges in the high and low side of the LED matrix due to multiplexing. There is a procedure for controlling this however many P6 panels are somewhat limited on this. This code base does not support much outside of lowering the refresh rate. Bleeding or ghosting increases as a function of refresh rate.

On some P6 panels there is optional resistor pads for installing pull down resistors on the high side. Adding a resistor like 10k will help some but not likely enough to remove this completely.

Would you like me to explain how it happens?

**Comment by daveythacher** [comment]
Another possible issue here is power supply and/or chain length. Make sure your power supply can carry the max load.

Make sure the clock speed is set low enough to handle a long chain. As the chain increases in length there can be artifacts due to small capacitors on the ribbon cable, PCB traces and IC pins. The net result of this is an attenuation of signal quality which can create artifacts periodically. There is no real solution in most cases other than lowering the clock speed or using multiple controllers.
